psl-core/src/test/java/org/linqs/psl/application/inference/MPEInferenceTest.java

```diff
@@ -195,4 +197,68 @@ public void testLogicalTautologyTrivial() {
         mpe.close();
         inferDB.close();
     }
+
+    /**
+    * Make sure that commitAtom flag works and all atoms are persisted in the database.
+    */
+    @Test
+    public void testCommitAtom() {
+        TestModel.ModelInformation info = TestModel.getModel();
+        Set<StandardPredicate> toClose = new HashSet<StandardPredicate>();
+        Database inferDB = info.dataStore.getDatabase(info.targetPartition, toClose, info.observationPartition);
+        InferenceApplication mpe = new MPEInference(info.model, inferDB);
+                
+        float preInferenceTotalValue = 0.0f;
+        for (RandomVariableAtom atom : inferDB.getAllGroundRandomVariableAtoms(info.predicates.get("Friends"))) {
+            preInferenceTotalValue += atom.getValue();
+        }
+
+        mpe.inference(true);
+        
+        mpe.close();
+        
+        // The database should be closed after inference, before reading in values again, to clear cache 
```

Comment:
"clear the cache".

Also if comments are getting long, try to break them on clauses.